### PR TITLE
Add audio support

### PR DESCRIPTION
**Pull request by kblok**
I've added audio support for this great game.

Basically we have an array of sounds (with a name) that can triggered from the engine using audioEngine.trigger.

I think that the showMessage function and the audioEngine trigger function could be replaced for a kind of producer-consumer pattern.

### CODE REVIEW

File: engine/behaviors.js [hunk 1]
```diff
@@ -123,6 +123,8 @@
 
 			function fireBullet(xDirection, yDirection, tFrame){
 		      //console.log('firing bullet', xDirection, yDirection);
+					LP.audioEngine.trigger("fire");
```
Comment (LucasMetal) [inline engine/behaviors.js:126]: Wonder how is the effect when the circleEnemy fires 18 bullets at the same time! Let's check it out...

File: engine/audioEngine.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+(function(LP) {
+
+  //Creates an audio element
+  function getSound(audioItem) {
+    var sound = document.createElement("audio");
```
Comment (LucasMetal) [inline engine/audioEngine.js:5]: Wonder if creating an element for each sound doesn't create too much work on the garbage collector or something. Anyway, we could optimize it later if required ;)

File: index.html [hunk 3]
```diff
@@ -32,6 +33,9 @@
     <div id="messages" style="margin-left:15%;float:left;"></div>
     <div id="stats" style='float:right;'></div>
   </div>
+  <div id="options" style="margin-top: 50px; float:left;">
```
Comment (LucasMetal) [inline index.html:36]: ha! With UI config and everything! Crack! jaja

**Comment by LucasMetal** [comment]
Totally agree on the pub-sub pattern, the game is still too green, but now it will have sound! Thanks!

File: engine/behaviors.js (hunk 1 above)
Comment (kblok) [inline engine/behaviors.js:126]: That's true, maybe this isn't the best place to trigger the "fire" sound

**Pull request merged**